packages/@uppy/xhr-upload/src/index.js

```diff
@@ -248,32 +248,26 @@ module.exports = class XHRUpload extends Plugin {
           this.uploaderEvents[file.id] = null
         }
 
-        if (opts.validateStatus(ev.target.status, xhr.responseText, xhr)) {
-          const body = opts.getResponseData(xhr.responseText, xhr)
-          const uploadURL = body[opts.responseUrlFieldName]
+        const body = opts.getResponseData(xhr.responseText, xhr)
+        const uploadURL = body[opts.responseUrlFieldName]
```

Comment:
If there was an error, there may not be any response data—this was why it was initially separate. There can't really be an `uploadURL` if the upload failed :) we can do a check like this though:
```suggestion
        const uploadURL = body != null ? body[opts.responseUrlFieldName] : null
```

Reply:
@goto-bus-stop Wouldn't there be an attempted upload url of sorts, the end point that failed?